src/sentry/south_migrations/0457_global_searches_data_migration.py

```diff
@@ -0,0 +1,1304 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+DEFAULT_SAVED_SEARCHES = [
+    {
+        'name': 'Unresolved Issues',
+        'query': 'is:unresolved',
+        'is_default': True
+    },
+    {
+        'name': 'Needs Triage',
+        'query': 'is:unresolved is:unassigned'
+    },
+    {
+        'name': 'Assigned To Me',
+        'query': 'is:unresolved assigned:me'
+    },
+    {
+        'name': 'My Bookmarks',
+        'query': 'is:unresolved bookmarks:me'
+    },
+    {
+        'name': 'New Today',
+        'query': 'is:unresolved age:-24h'
+    },
+]
+
+
+class Migration(DataMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = True
+
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
+
+    def _forwards(self, orm):
+        "Write your forwards methods here."
+        # Note: Don't use "from appname.models import ModelName".
+        # Use orm.ModelName to refer to models in this application,
+        # and orm['appname.ModelName'] for models in other applications.
+        SavedSearch = orm['sentry.SavedSearch']
+
+        for search in RangeQuerySetWrapperWithProgressBar(
+                SavedSearch.objects.filter(is_global__isnull=True)
+        ):
+            search.is_global = False
+            search.save()
+
+        default_searches = []
+        for search in DEFAULT_SAVED_SEARCHES:
+            default_searches.append(
+                SavedSearch(
+                    name=search['name'],
+                    query=search['query'],
+                    is_global=True,
+                )
+            )
+        SavedSearch.objects.bulk_create(default_searches)
+
+    def backwards(self, orm):
+        "Write your backwards methods here."
+        # These will be the only rows with a null `project_id`, so we can safely
+        # make the column `not null` after deleting them.
+        SavedSearch = orm['sentry.SavedSearch']
+        SavedSearch.objects.filter(is_global=True).delete()
```

Comment:
we don't usually implement these since the only time we'd ever use them is while we're working on the forwards migration